include/rocksdb/transaction_log.h

```diff
@@ -56,8 +56,25 @@ class LogFile {
 };
 
 struct BatchResult {
-  SequenceNumber sequence = 0;
-  std::unique_ptr<WriteBatch> writeBatchPtr;
+    SequenceNumber sequence = 0;
+    std::unique_ptr<WriteBatch> writeBatchPtr;
+
+    BatchResult() {
+    }
+
+    BatchResult(const BatchResult&) = delete;
+
+    BatchResult& operator=(const BatchResult&) = delete;
```

Comment:
@igorcanadi Not really, the semantics of the object and how it was and being used does not suggest that we need two functional copies of it due to the fact that the std::unique_ptr also has copy and assignment deleted. So I chose to replicate those because it yields fast return from functions and container insertion automatically w/o std::move.